pypose/module/icp.py

```diff
@@ -0,0 +1,74 @@
+import torch
+from . import EPnP
+from .. import LieTensor
+class ICP(torch.nn.Module):
+    r'''
+    Iterative Closest Point (ICP) using Singular Value Decomposition (SVD).
+    Args:
+        steplim: the max step number
+        tol: the tolerance of error for early stopping
+        matched: whether the input points set and the target points set have been matched
+    '''
+
+    def __init__(self, steplim=200, tol=0.0001, init_transform =None):
+        super().__init__()
+        assert type(init_transform) == LieTensor
+        self.steplim = steplim
+        self.tol = tol
+        self.init_transform = init_transform
+
+    def forward(self, p1, p2):
+        temppc = p1.clone()
+        iter = 0
+        err = None
+        if (self.init_transform != None):
+            temppc = self.init_transform.Act(temppc)
+        while iter <= self.steplim:
+            iter += 1
+            neighbors = self._k_nearest_neighbor(temppc, p2)
+            errnew = torch.mean(neighbors.values, dim=-1)
+            T = EPnP._points_transform(temppc, p2[:, neighbors.indices[-1],:].squeeze(-2))
```

Comment:
I suggest defining an independent function for `points_transform` and expose to users, then both `EPnP`and `ICP` call this new function.

Reply:
Should move `points_transform` to `pypose.geometry`?

Reply:
yes, we can do that

Reply:
> Should move `points_transform` to `pypose.geometry`?

Hi Zitong, will the points_transformation function be moved to the geometry module in this pull request? If so, would you prefer to make the change yourself or would you like our assistance with the modification?

Reply:
You may move that function directly in this PR